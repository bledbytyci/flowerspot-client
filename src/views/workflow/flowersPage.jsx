import React, {Component} from 'react';
import { connect } from 'react-redux';
import FlowerActionCreators from '../../flower/flowerActionCreators';
import FlowerSelectors from '../../flower/flowerSelectors';
import '../../styles/flower.css';
import FlowerCard from './forms/flower/flowerCard.jsx';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import { Row } from 'react-bootstrap';
import AuthSelectors from '../../auth/authSelectors';
import AuthActionCreators from '../../auth/authActionCreators';

const mapStoreToProps = store => {
	return {
		flowers: FlowerSelectors.getFlowers(store),
		isLoggedIn: AuthSelectors.isLoggedIn(store)
	}
}

const mapDispatchToProps = dispatch => {
	return {
		getFlowers: () => dispatch(FlowerActionCreators.getFlowers()),
		checkUserIsLoggedIn: () => dispatch(AuthActionCreators.checkUserIsLoggedIn())
	}
}

class FlowersPage extends Component {
	constructor(props){
		super(props)
	}

	componentDidMount() {
		this.props.getFlowers();
	}

	static get propTypes() {
		return {
			checkUserIsLoggedIn: PropTypes.func,
			getFlowers: PropTypes.func,
			flowers: PropTypes.instanceOf(List),
			isLoggedIn: PropTypes.bool
		}
	} 

	render() {
		const {flowers, isLoggedIn} = this.props;
		return(
			<div className="container-fluid flower-container">
				<Row>
					{flowers.map((flower, key) => (
						<FlowerCard flower={flower} key={key} isLoggedIn={isLoggedIn} />
					))}
				</Row>
			</div>
		)
	}
}

export default connect(mapStoreToProps, mapDispatchToProps)(FlowersPage);