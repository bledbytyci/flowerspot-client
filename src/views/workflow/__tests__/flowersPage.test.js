import React from 'react';
import { FlowersPage } from '../flowersPage.jsx';
import { shallow} from 'enzyme';
import { List } from 'immutable';
import toJson from 'enzyme-to-json';

jest.unmock('../flowersPage.jsx');
jest.unmock('../forms/flower/flowerCard.jsx');
jest.unmock('../../../flower/flower.js');

describe('FlowersPage component', () => {
    it('component is being rendered', () => {
		const props = {
			flowers: List(),
			isLoggedIn: false,
			getFlowers: jest.fn()
		}
		const flowersPageComponent = shallow(<FlowersPage {...props} />);
        expect(flowersPageComponent).not.toBe(null);
	});
});

describe('snapshots', () => {
	it('App snapshot', () => {
		const props = {
			flowers: List(),
			isLoggedIn: false,
			getFlowers: jest.fn()
		}
		const flowersPageComponent = shallow(<FlowersPage {...props} />);
		expect(toJson(flowersPageComponent)).toMatchSnapshot();
	})
})