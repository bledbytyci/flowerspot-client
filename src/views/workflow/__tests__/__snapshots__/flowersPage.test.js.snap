// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshot 1`] = `
<Fragment>
  <Connect(Header) />
  <div
    className="container-fluid flower-container"
  >
    <Row
      noGutters={false}
    />
  </div>
</Fragment>
`;
